tee-worker/scripts/litentry/release/deploy.sh

```diff
@@ -1,736 +1,555 @@
+      args=
+      for flag in $flags; do
+        args+=" $flag"
+      done
+      args+=" run"
+      for subcommand_flag in $subcommand_flags; do
+        args+=" $subcommand_flag"
+      done
+      sed -i "s/ARGS/$args/" worker$i.service
+    done
+    rm worker.service
+    sudo cp *.service -f /etc/systemd/system/
+    rm *.service
+    sudo systemctl daemon-reload
+    echo "Done, please check files under /etc/systemd/system/"
+    echo "Restart the services to take effect"
+}
 
-  cd "$ROOTDIR" || exit
-  PARACHAIN_ID=$(grep DEFAULT_PARA_ID node/src/chain_specs/$CHAIN.rs  | grep u32 | sed 's/.* = //;s/\;//')
-  export PARACHAIN_ID
-  echo "Parachain ID: $PARACHAIN_ID"
-
-  echo "Could not find Polkadot Binary, Downloading now"
-  echo "no polkadot binary provided, download now ..."
-  url="https://github.com/paritytech/polkadot/releases/download/v0.9.39/polkadot"
-  POLKADOT_BIN="$TMPDIR/polkadot"
-  wget -O "$POLKADOT_BIN" -q "$url"
-  chmod a+x "$POLKADOT_BIN"
-
-  if [ ! -s "$POLKADOT_BIN" ]; then
-    echo "$POLKADOT_BIN is 0 bytes, download URL: $url"
-    exit 1
+function build_worker {
+  echo "Building worker ..."
+  cd $ROOTDIR/tee-worker/ || exit 
+  if [ "$PRODUCTION" = true ]; then
+    # we will get an error if SGX_COMMERCIAL_KEY is not set for prod
+    SGX_PRODUCTION=1 make
+  else
+    # use SW mode for dev
+    SGX_MODE=SW make
   fi
+}
 
-  if ! "$POLKADOT_BIN" --version &> /dev/null; then
-    echo "Cannot execute $POLKADOT_BIN, wrong executable?"
-    usage
-    exit 1
-  fi
+# TODO: take github rev into consideration
+function build {
+  if [ "$BUILD" = true ]; then
+    echo "Building the parachain and worker binaries ..."
+
+    # download polkadot
+    echo "Downloading polkadot binary ..."
+    url="https://github.com/paritytech/polkadot/releases/download/v0.9.39/polkadot"
+    polkadot_bin="$PARACHAIN_BASEDIR/polkadot"
+    wget -O "$polkadot_bin" -q "$url"
+    chmod a+x "$polkadot_bin"
+    if [ ! -s "$polkadot_bin" ]; then
+      echo "$polkadot_bin is 0 bytes, download URL: $url" && exit 1
+    fi
+    if ! "$polkadot_bin" --version &> /dev/null; then
+      echo "Cannot execute $polkadot_bin, wrong executable?" && exit 1
+    fi
 
-  echo "Fething Litentry Collator Binary"
-  if [ "$COPY_FROM_DOCKER" = true ]; then
-    PARACHAIN_BIN="$ROOTDIR/litentry-collator"
-  else
-    PARACHAIN_BIN="$ROOTDIR/target/release/litentry-collator"
+    # pull or build parachain
+    if [ "$COPY_FROM_DOCKER" = true ]; then
+      echo "Pulling binary from $DOCKER_IMAGE ..."
+      docker pull "$DOCKER_IMAGE"
+      docker cp "$(docker create --rm $DOCKER_IMAGE):/usr/local/bin/litentry-collator" "$PARACHAIN_BASEDIR"
+    else
+      echo "Building parachain binary ..."
+      cd "$ROOTDIR" || exit
+      if [ "$PRODUCTION" = true ]; then
+        cargo build --locked --profile production
+      else
+        pwd
+        make build-node
```

Comment:
Shouldn't we have a flag here to build for production, I think `make build-node` only builds with `tee-dev` feature. We will need without `tee-dev` feature for production.

Reply:
Just a few lines above :) Line #348

Reply:
Ah my bad 😅